yt_dlp/extractor/rcs.py

```diff
@@ -216,32 +217,42 @@ def _real_extract(self, url):
         video_data = None
         # look for json video data url
         json = self._search_regex(
-            r'''(?x)var url\s*=\s*["']((?:https?:)?
-                //video\.rcs\.it
-                /fragment-includes/video-includes/.+?\.json)["'];''',
-            page, video_id, default=None)
+            r'''(?x)url\s*=\s*(["'])
+            (?P<url>
+                (?:https?:)?//video\.rcs\.it
+                /fragment-includes/video-includes/.+?\.json
+            )\1;''',
+            page, video_id, group='url', default=None)
         if json:
             if json.startswith('//'):
                 json = 'https:%s' % json
             video_data = self._download_json(json, video_id)
 
         # if json url not found, look for json video data directly in the page
         else:
+            # RCS normal pages and most of the embeds
             json = self._search_regex(
                 r'[\s;]video\s*=\s*({[\s\S]+?})(?:;|,playlist=)',
                 page, video_id, default=None)
-            if json:
-                video_data = self._parse_json(
-                    json, video_id, transform_source=js_to_json)
-            else:
+            if not json and 'video-embed' in url:
+                # RCSEmbeds_2 test
```

Comment:
What is `RCSEmbeds_2`? If u r writing comments, make sure someone new looking at the code understands it. Otherwise it only ends up being more confusing

Reply:
it the corresponding test that "triggers" this check.

python test\test_download.py TestDownload.test_**RCSEmbeds_2**